compiler/record-minmax/src/HDF5Importer.h

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (c) 2020 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef __RECORD_MINMAX_HDF5IMPORTER_H__
+#define __RECORD_MINMAX_HDF5IMPORTER_H__
+
+#include <luci_interpreter/core/Tensor.h>
+
+#include <H5Cpp.h>
+
+using DataType = luci_interpreter::DataType;
+
+namespace record_minmax
+{
+
+// HDF5Importer reads an input data saved in the hdf5 file in the given path
+// The hierarchy of the hdf5 file is as follows.
+// Group "/"
+//  > Group "value"
+//    > Group <record_idx>
+//      > Dataset <input_idx>
+// record_idx : index of the record (dataset file can contain multiple records)
+// input_idx : index of the input (DNN model can have multiple inputs)
+// Ex: the j'th input of the i'th record can be accessed by "/value/i/j"
+class HDF5Importer
+{
+public:
+  HDF5Importer(const std::string &path) : _file{path, H5F_ACC_RDONLY}
+  {
+    _value_grp = _file.openGroup("value");
```

Comment:
(Q) (not knowing about HDF5), is it OK if `value` is not found in group? will it throw? or return is invalid?

Reply:
If the value is not found, H5::FileIException is thrown as below.
(I tried to open a group named "asdf" which does not exist)

```
~/ONE$ build/compiler/record-minmax/record-minmax build/compiler/record-minmax-conversion-test/Add_000.circle build/compiler/record-minmax-conversion-test/Add_000.tflite.input.h5 out.circle
HDF5-DIAG: Error detected in HDF5 (1.10.0-patch1) thread 140459280253376:
  #000: ../../../src/H5G.c line 474 in H5Gopen2(): unable to open group
    major: Symbol table
    minor: Can't open object
  #001: ../../../src/H5Gint.c line 289 in H5G__open_name(): group not found
    major: Symbol table
    minor: Object not found
  #002: ../../../src/H5Gloc.c line 430 in H5G_loc_find(): can't find object
    major: Symbol table
    minor: Object not found
  #003: ../../../src/H5Gtraverse.c line 869 in H5G_traverse(): internal path traversal failed
    major: Symbol table
    minor: Object not found
  #004: ../../../src/H5Gtraverse.c line 641 in H5G_traverse_real(): traversal operator failed
    major: Symbol table
    minor: Callback failed
  #005: ../../../src/H5Gloc.c line 385 in H5G_loc_find_cb(): object 'asdf' doesn't exist
    major: Symbol table
    minor: Object not found
terminate called after throwing an instance of 'H5::FileIException'
```

Reply:
(optional) as like previous one, it is better not to add codes that throw in CTOR...

Reply:
Changed the function. Thanks.